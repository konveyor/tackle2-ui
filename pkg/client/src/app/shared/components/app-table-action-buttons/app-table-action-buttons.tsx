import React from "react";
import { useTranslation } from "react-i18next";
import { Button, Flex, FlexItem } from "@patternfly/react-core";

export interface AppTableActionButtonsProps {
  onEdit: () => void;
  onDelete: () => void;
}

export const AppTableActionButtons: React.FC<AppTableActionButtonsProps> = ({
  onEdit,
  onDelete,
}) => {
  const { t } = useTranslation();

  return (
    <Flex>
      <FlexItem align={{ default: "alignRight" }}>
        <Button aria-label="edit" variant="secondary" onClick={onEdit}>
          {t("actions.edit")}
        </Button>
      </FlexItem>
      <FlexItem>
        <Button aria-label="delete" variant="link" onClick={onDelete}>
          {t("actions.delete")}
        </Button>
      </FlexItem>
    </Flex>
  );
};
