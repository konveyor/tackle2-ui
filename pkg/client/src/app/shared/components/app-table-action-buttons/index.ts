export { AppTableActionButtons } from "./app-table-action-buttons";
