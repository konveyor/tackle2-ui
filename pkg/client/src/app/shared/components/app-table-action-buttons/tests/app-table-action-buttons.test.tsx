import React from "react";
import { mount, shallow } from "enzyme";

import { AppTableActionButtons } from "../app-table-action-buttons";

describe("AppTableActionButtons", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <AppTableActionButtons onEdit={jest.fn()} onDelete={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Callback edit", () => {
    const callbackSpy = jest.fn();
    const wrapper = mount(
      <AppTableActionButtons onEdit={callbackSpy} onDelete={jest.fn()} />
    );

    wrapper.find('button[aria-label="edit"]').simulate("click");

    expect(callbackSpy).toHaveBeenCalledTimes(1);
  });

  it("Callback delete", () => {
    const callbackSpy = jest.fn();
    const wrapper = mount(
      <AppTableActionButtons onEdit={jest.fn()} onDelete={callbackSpy} />
    );

    wrapper.find('button[aria-label="delete"]').simulate("click");

    expect(callbackSpy).toHaveBeenCalledTimes(1);
  });
});
