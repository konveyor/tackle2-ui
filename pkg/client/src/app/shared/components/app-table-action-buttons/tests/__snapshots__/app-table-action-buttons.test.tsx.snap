// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTableActionButtons Renders without crashing 1`] = `
<Flex>
  <FlexItem
    align={
      Object {
        "default": "alignRight",
      }
    }
  >
    <Button
      aria-label="edit"
      onClick={[MockFunction]}
      variant="secondary"
    >
      actions.edit
    </Button>
  </FlexItem>
  <FlexItem>
    <Button
      aria-label="delete"
      onClick={[MockFunction]}
      variant="link"
    >
      actions.delete
    </Button>
  </FlexItem>
</Flex>
`;
